web/packages/teleport/src/Main/Main.tsx

```diff
@@ -117,6 +138,22 @@ export function Main(props: MainProps) {
     !!props.Questionnaire
   );
 
+  const [previousManagementRoute, setPreviousManagementRoute] = useState('');
+
+  const handleLocationChange = useCallback(
+    (next: history.Location<unknown> | Location) => {
+      const category = getCategoryForRoute(features, next);
+      if (category && category === NavigationCategory.Management) {
+        setPreviousManagementRoute(next.pathname);
+      }
+    },
+    [features]
+  );
+
+  useEffect(() => {
+    return history.listen(handleLocationChange);
+  }, [history, handleLocationChange]);
```

Comment:
Why here instead of in `TopBar`?

Reply:
Good question. It was here because I was originally passing things to navigation + topbar, but after refactoring a lot, i ended up not needing to pass to navigation. thanks for catching it. moved to topbar and its even cleaner now